src/format/parse.rs

```diff
@@ -574,11 +573,19 @@ fn parse_rfc3339_relaxed<'a>(parsed: &mut Parsed, mut s: &'a str) -> ParseResult
         None => return Err(TOO_SHORT),
     };
 
-    match parse_internal(parsed, s, TIME_ITEMS.iter()) {
-        Err((s, e)) if e.0 == ParseErrorKind::TooLong => Ok((s, ())),
-        Err((_s, e)) => Err(e),
-        Ok(s) => Ok((s, ())),
-    }
+    s = match parse_internal(parsed, s, TIME_ITEMS.iter()) {
+        Err((s, e)) if e.0 == ParseErrorKind::TooLong => s,
+        Err((_s, e)) => return Err(e),
+        Ok(_) => return Err(NOT_ENOUGH),
+    };
+    s = s.trim_start();
+    let (s, offset) = if s.starts_with("UTC") || s.starts_with("utc") {
```

Comment:
I am not sure which change you mean. Moving this code from `scan::timezone_offset` to `parse_rfc3339_relaxed`?

It seems like a hack to me that if we set the `allow_zulu` flag to parse `Z` we also accept the time zone abbreviation `UTC`. That is what the `TimezoneName` item is for.

I'll add some comments.

Reply:
I guess the change in general? You're moving support for the UTC offset suffix from `timezone_offset()` to `parse_rfc3339_relaxed()`. As far as I can tell, this makes our parsing more strict, so some parses that used to work will now fail. Why do you want to make this change?

Reply:
> As far as I can tell, this makes our parsing more strict, so some parses that used to work will now fail.

I have to admit I forgot about it (I wrote the first version end of April). It would only change how the following two items parse:
```rust
    /// Offset from the local time to UTC (`+09:00` or `-04:00` or `Z`).
    ///
    /// In the parser, the colon can be omitted and/or surrounded with any amount of whitespace,
    /// and `Z` can be either in upper case or in lower case.
    /// The offset is limited from `-24:00` to `+24:00`,
    /// which is the same as [`FixedOffset`](../offset/struct.FixedOffset.html)'s range.
    TimezoneOffsetColonZ,
    /// Same as [`TimezoneOffsetColonZ`](#variant.TimezoneOffsetColonZ) but prints no colon.
    /// Parsing allows an optional colon.
    TimezoneOffsetZ,
```

Both are not currently exposed as a formatting specifier, except as part of `%+`. And they don't mention anything about accepting "UTC". So I thought the impact would be basically none.

It was added in https://github.com/chronotope/chrono/pull/378 for `%+`, with no mention that it would make `TimezoneOffsetColonZ` and `TimezoneOffsetZ` accept more inputs.

But good point, that is important!

Reply:
> What about mixed-case `UtC` or `Utc`?

Good catch, I did not think about this. I have something against case-insensitive compares that accept things like `UtC`. But it was accepted before. What do you think?

Reply:
Done.